lib/internal/main/eval_string.js

```diff
@@ -25,9 +27,31 @@ const print = getOptionValue('--print');
 const loadESM = getOptionValue('--import').length > 0;
 if (getOptionValue('--input-type') === 'module')
   evalModule(source, print);
-else
+else {
+  // For backward compatibility, we want the identifier crypto to be the
+  // `node:crypto` module rather than WebCrypto.
+  const isUsingCryptoIdentifier =
+                             getOptionValue('--experimental-global-webcrypto') &&
+                             RegExpPrototypeExec(/\bcrypto\b/, source) !== null;
+  const shouldDefineCrypto = isUsingCryptoIdentifier && internalBinding('config').hasOpenSSL;
+
+  if (isUsingCryptoIdentifier && !shouldDefineCrypto) {
+    // This is taken from `addBuiltinLibsToObject`.
+    const object = globalThis;
+    const name = 'crypto';
+    const setReal = (val) => {
+      // Deleting the property before re-assigning it disables the
+      // getter/setter mechanism.
+      delete object[name];
+      object[name] = val;
+    };
+    ObjectDefineProperty(object, name, { __proto__: null, set: setReal });
+  }
   evalScript('[eval]',
-             source,
+             shouldDefineCrypto ? (
+               print ? `let crypto=require("node:crypto");{${source}}` : `(crypto=>{{${source}}})(require('node:crypto'))`
```

Comment:
```suggestion
               print ? `var crypto=require("node:crypto");{${source}}` : `(crypto=>{{${source}}})(require('node:crypto'))`
```

Should use `var` to allow users to redefine the variable?

Reply:
No, it would try to define it as a global, or rather trying to overwrite the global (and fail because there's no setter):

```console
$ node -e 'var myVar = {};console.log(Object.hasOwn(global, "myVar"))'
true
$ node -e 'let myVar = {};console.log(Object.hasOwn(global, "myVar"))'
false
$ node -e 'global.myVar = "global"; var myVar = "var"; console.log(global.myVar)'
var
$ node -e 'global.myVar = "global"; let myVar = "let"; console.log(global.myVar)'
global
```

Reply:
Well, that doesn't seem wrong. Other builtin modules are also defined as globals.

Reply:
`globalThis.crypto` is the WebCrypto API, and we are defining a local identifier `crypto` to access `node:crypto`.

Reply:
So `node -p "var crypto = require('node:crypto'); crypto;"` is going to throw a SyntaxError? I guess that's fine, but I wanted to clarify that.

Reply:
Yes, correct. Note that we could also add another check to address this specific use case, that doesn't seem too complicated.

Reply:
Hum actually, that would mean either modify the user code, or remove `globalThis.crypto`, and I feel like it would breaking user expectations in both cases... Maybe best to leave it as a known limitation, wdyt?

Reply:
sgtm